src/irr/asset/CImageLoaderKTX.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+MIT License
+Copyright (c) 2019 Achal Pandey
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include "CImageLoaderKTX.h"
+
+#ifdef _IRR_COMPILE_WITH_KTX_LOADER_
+
+#ifdef _IRR_COMPILE_WITH_GLI_
+#include "gli/gli.hpp"
+#endif
+
+#include "irr/asset/ICPUTexture.h"
+#include "irr/asset/CImageData.h"
+#include "COpenGLTexture.h"
+#include "CReadFile.h"
+#include "os.h"
+
+namespace irr
+{
+namespace asset
+{
+
+static bool checkKTXSignature(unsigned char* buffer)
+{
+    // Unique KTX identifier as per the Khronos KTX specification
+    // https://www.khronos.org/opengles/sdk/tools/KTX/file_format_spec/#2.1
+    unsigned char signature[12] = { 0xAB, 0x4B, 0x54, 0x58, 0x20, 0x31, 0x31, 0xBB, 0x0D, 0x0A, 0x1A, 0x0A };
+    
+    for (unsigned int i = 0; i < 12; i++)
+    {
+        if (buffer[i] != signature[i])
+            return false;
+    }
+    return true;
+}
+
+//! returns true if the file maybe is able to be loaded by this class
+bool CImageLoaderKTX::isALoadableFileFormat(io::IReadFile* _file) const
+{
+#ifdef _IRR_COMPILE_WITH_GLI_
+    if (!_file)
+        return false;
+
+    const size_t prevPos = _file->getPos();
+    
+    unsigned char buffer[12];
+
+    // Read the first few bytes of the KTX _file
+    if (_file->read(buffer, 12) != 12)
+    {
+        _file->seek(prevPos);
+        return false;
+    }
+
+    _file->seek(prevPos);
+
+    // Check if it really is a KTX _file
+    return checkKTXSignature(buffer);
```

Comment:
gli loads both DDS and KTX, you could nuke our DDS loader and have this class handle both KTX and DDS.